src/rgw/rgw_sal_daos.cc

```diff
@@ -2866,7 +2867,12 @@ std::unique_ptr<Object> DaosStore::get_object(const rgw_obj_key& k) {
 int DaosStore::get_bucket(const DoutPrefixProvider* dpp, User* u,
                           const rgw_bucket& b, std::unique_ptr<Bucket>* bucket,
                           optional_yield y) {
-  ldpp_dout(dpp, 20) << "DEBUG: get_bucket1: User" << u->get_id() << dendl;
+  if (u) {
+    ldpp_dout(dpp, 20) << "DEBUG: get_bucket1: User: " << u->get_id() << dendl;
+  }
+  else {
+    ldpp_dout(dpp, 20) << "DEBUG: get_bucket1: User: (nullptr)" << dendl;
+  }
```

Comment:
A ternary operator could make this less redundant: `(u? u->get_id(): "nullptr")`

Reply:
the operation u-get_id() doesn't return a string, so you can't

Reply:
I could add and operator<< that takes a User* which would improve the readability